app/models/test_track/session.rb

```diff
@@ -11,34 +11,25 @@ def initialize(controller)
   def manage
     yield
   ensure
-    manage_cookies!
-    manage_response_headers!
+    if visitor.id_loaded?
+      manage_cookies!
+      manage_response_headers!
+    end
```

Comment:
These are the operations that aren't worth doing if we don't actually use the visitor. Only possible bad outcome I can imagine from this would be that the cookie never ever gets an updated expiry timestamp if we never use TestTrack in an authenticated controller *and* the user never experiences an unauthenticated page for a whole year. Not likely.